src/module_2/EDA.md

```diff
@@ -0,0 +1,324 @@
+
+
+```python
+print("Estadísticas de 'normalised_price':")
+print(df['normalised_price'].describe())
+
+print("\nEstadísticas de 'discount_pct':")
+print(df['discount_pct'].describe())
+```
+
+Evaluamos los casos en los que tenemos un descuento  superior a 1 e inferior a 0:
+
+
+```python
+unusual_dicounts_df_negative = df[(df['discount_pct'] < 0)]
+print(len(unusual_dicounts_df_negative.index))
+
+unusual_dicounts_df_up_to_1 = df[(df['discount_pct'] > 1)]
+print(len(unusual_dicounts_df_up_to_1.index))
+
+```
+
+Vemos que hay 1587 casos con un descuento negativo y 58582 casos con un descuento superior a 1. Habría que preguntar al negocio si esto realmente tiene sentido.
+
+**3. Características del usuario y comportamiento de compra:** 
+
+- user_order_seq: Secuencia de orden del usuario. (int64)
+- outcome: Indica si el producto fue comprado o no en ese pedido. (float64)
+- ordered_before: Indica si el producto fue ordenado previamente. (float64)
+- abandoned_before: Indica si el producto fue abandonado anteriormente. (float64)
+- active_snoozed: Indica si el producto está activo o en modo de espera. (float 64)
+- set_as_regular: Indica si el producto se estableció como regular. (float 64)
+
+
+```python
+print("Número secuencia de orden del usuario:", len(df['user_order_seq'].unique()))
+print(df['user_order_seq'].value_counts())
+```
+
+La variable 'user_order_seq' está directamente relacionada con el usuario e indica el número de pedidos que ha realizado cada uno de los usuarios. Se observa una clara tendencia decreciente a medida que aumenta el número de orden, concentrándose la mayor parte de usuarios en 2 pedidos.
+
+Visualizo esa tendencia decreciente de la secuencia de orden de los usuarios:
+
+
+```python
+unique_seq_lengths = len(df['user_order_seq'].unique())
+seq_count = df['user_order_seq'].value_counts()
+
+plt.bar(seq_count.index, seq_count.values, color='blue', alpha=0.7)
+
+plt.xlabel('Número de Secuencias de Orden')
+plt.ylabel('Número de Usuarios')
+plt.title(f'Histograma de Número de Secuencias de Orden por Usuario (Total: {unique_seq_lengths} longitudes únicas)')
+
+plt.show()
+```
+
+Ahora voy a ver cómo se comportan el resto de variables binarias. Están codificadas todas ellas como float pero, al tratase de variables binarias, por simplicidad, se van a tratar como int.
+
+
+```python
+binary_variables = ['outcome', 'ordered_before', 'abandoned_before', 'active_snoozed', 'set_as_regular']
+df[binary_variables] = df[binary_variables].astype(int)
+
+for var in binary_variables:
+    print(f'Valores únicos y frecuencia para {var}:\n{df[var].value_counts()}\n')
+```
+
+Se observa que, del total de datos, únicamente 33000 registros aproximadamente acaban comprando el producto y que, la mayoría de los productos no se establecen como regulares, ni fueron abandonados previamente ni están en espera ni fueron ordenados previamente.
```

Comment:
But why?! The key here is to understand that the dataset contains every product for each order, so it's logical that those binary variables are very unbalanced. Isn't it?